content/reliability/docs/architecting-for-static-stability.md

```diff
@@ -0,0 +1,19 @@
+
+For an isolated availability event, pods should be terminated and rescheduled elsewhere to restore the desired available replicas of its owning controller. Kubernetes achieves this through [taint based eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions). After 40 seconds without a heartbeat, the node lifecycle controller applies a NoExecute taint to the node. By default, pods tolerate this NoExecute taint for 300 seconds, before they are terminated by the pod lifecycle controller. The owning controller launches a new pod, which is then scheduled to a healthy node.
+
+This behavior works well for isolated availability events, but what happens during a correlated event, such as an availability zone or regional power failure? When 55% of the nodes in an availability zone fail to send a heartbeat, the node lifecycle controller determines that the cluster is in a state of partial disruption and halts taint based eviction. Once the number of available nodes returns below this threshold, taint based eviction is restored. If an entire availability zone goes down, why doesn’t Kubernetes to taint all of the nodes and migrate the workloads to another availability zone? Architecting for availability zone evacuation (e.g. running at a high utilization with the plan to launch new capacity during an AZ failure) contains multiple bad assumptions that risk application availability.
+
+First, is the assumption that capacity will be available during the outage. If the event is correlated, workloads across accounts may be impacted. If each of these workloads evacuate at the same time, this will result in a large spike in demand for capacity in the other availability zones. In many cases, this can be absorbed by large capacity pools in EC2, but risks increase for higher demand instance types and smaller regions.
+
+Second, is the assumption that control plane orchestration systems are not impacted by the correlated outage. While these systems are highly available, the number of systems involved in failover spans many AWS services, core Kubernetes components, and user installed Kubernetes operators. Evacuation results in a large spike in traffic across all of these systems, and correlated impact to any one of these systems can cause an evacuation strategy to fail.
+
+Third, is the introduction of bi-modal system behavior. Availability zone evacuation is a rare event, outside of the standard operations of the system. This mechanism is expensive to test, and thus, rarely tested. Even if it has been tested, changes to the system over time may violate original assumptions, which won’t be discovered until it’s too late.
```

Comment:
```suggestion
Third is the introduction of bi-modal system behavior. Availability Zone evacuation is a rare event outside of the standard operations of the system. This mechanism is expensive to test, and thus, rarely tested. Changes to the system over time may violate original assumptions and test results. The impact of which won’t surface until it’s too late.
```

Reply:
I don't understand the reference to "bimodal system behavior".

Reply:
This was beaten into me by multiple EKS PEs like Gus Lees and Jacob Gabrielson. Does the paragraph not do justice to the concept?